### PR TITLE
BoardController refactor, position fix

### PR DESCRIPTION
**Pull request by agawda**
Refactoring of the Board Controller - javafx class.

### CODE REVIEW

File: client/src/main/java/com/blackship/battlesheep/fx/controllers/SalvoHandler.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        enemyHitPositions = new HashSet<>();
+    }
+
+    public boolean encounterSalvos() throws IOException, ClassNotFoundException {
+
+        PacketMove packet = (PacketMove) appClientCommunicationHandler.read();
+
+        if (((Packet)packet).getPacketType() == PacketType.MOVE) {
+            enemyHitPositions.addAll(packet.getPositions());
+//            playerHitPositions.addAll(packet.getPlayerPositions());
```
Comment (arturAR) [inline client/src/main/java/com/blackship/battlesheep/fx/controllers/SalvoHandler.java:40]: 👃

File: server/src/main/java/com/blackship/battlesheep/AppServer.java [hunk 2]
```diff
@@ -16,7 +16,7 @@ public static void main(String[] args) throws IOException, ClassNotFoundExceptio
         new ServerCommunicationHandler(
                 new AppServerSocket(8096))
                 .acceptClients()
-                .sendHardcodedBoardsToClients()
+                //.sendHardcodedBoardsToClients()
```
Comment (arturAR) [inline server/src/main/java/com/blackship/battlesheep/AppServer.java:19]: 👃

**Review by arturAR — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        updateEnemyBoard(salvoHandler.getEnemyHitPositions());
+//        updatePlayerBoard(salvoHandler.getPlayerHitPositions());
+        enemyMastPositions
+                .stream()
+                .filter(data -> !data.isDisabled())
+                .forEach(data -> data.setStyle(ButtonUtils.defaultButtonColorStyle()));
+    }
+
+    public void feedCannon(int cannonBall) {
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java:62]: Awesome name!

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        currentSalvo.add(cannonBall);
+        if (tempCannonsAmount == 0) {
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java:64]: Magic Numbers.

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+//    private void updatePlayerBoard(Set<Integer> positions) {
+//        positions.forEach(data -> {
+//            Label label = new Label();
+//            label.setText("X");
+//            int[] twoDimensionalBoardPosition = PositionUtils.calculateFromOneDimensionalPosition(data - 1, BOARD_SIZE);
+//            playerGridPane.add(label, twoDimensionalBoardPosition[X], twoDimensionalBoardPosition[Y]);
+//        });
+//    }
+
+    //TODO Solve magic number bug(?)
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java:83]: :+1:

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            salvoHandler.fireSalvoCannonade(currentSalvo);
+            currentSalvo.clear();
+            tempCannonsAmount = 4;
+            boardViewUpdaterListener.update("...!!!!FIRE!!!!...\n");
+        } else {
+            tempCannonsAmount--;
+        }
+    }
+
+//    private void updatePlayerBoard(Set<Integer> positions) {
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java:74]: Remove it.

File: client/src/main/java/com/blackship/battlesheep/fx/utils/ButtonUtils.java [hunk 7]
```diff
@@ -29,4 +33,12 @@ public static Button createStyledButton(int value) {
         button.setMinHeight(MIN_BUTTON_HEIGHT);
         return button;
     }
+
+    public static List<Button> generateButtons(int boardSize) {
+        List<Button> buttons = new ArrayList<>();
+        IntStream
+                .rangeClosed(1, boardSize * boardSize)
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/fx/utils/ButtonUtils.java:40]: Magic Number. For example ```private final int START_POSITION```, maybe not the best idea.

**Review by mhbratek — APPROVED** [review]

**Pull request merged**